src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -53,6 +59,13 @@
       SmtpCommand.QUIT,
       SmtpCommand.NOOP);
 
+  private static final Supplier<Executor> SHARED_DEFAULT_EXECUTOR = Suppliers.memoize(SmtpSession::getSharedExecutor);
+
+  private static ExecutorService getSharedExecutor() {
+    ThreadFactory threadFactory = new ThreadFactoryBuilder().setDaemon(true).setNameFormat("niosmtpclient-%d").build();
+    return Executors.newCachedThreadPool(threadFactory);
```

Comment:
Ah, that's fine then. Core of 0 has weird semantics w.r.t. queuing but combined with a SynchronousQueue (as cachedPool does) I think we're good to go